checker.go

```diff
@@ -269,6 +304,50 @@ func (c *hasLenChecker) Check(got interface{}, args []interface{}, note func(key
 	return nil
 }
 
+// Satisfies is a Checker checking that the provided value, when used as
+// argument of the provided function, causes the function to return true.
+// The function must be of type func(T) bool where T is the type of got.
+// For instance:
+//
+//     c.Assert(os.ErrNotExist, qt.Satisfies, os.IsNotExist)
+//     c.Assert(42, qt.Satisfies, func(v int) bool { return v == 42 })
+//
+var Satisfies Checker = &satisfiesChecker{
+	info: newInfo("satisfies", "arg", "function"),
+}
+
+type satisfiesChecker struct {
+	info
+}
+
+// Check implements Checker.Check by checking that args[0](got) == true.
+func (c *satisfiesChecker) Check(got interface{}, args []interface{}, note func(key, value string)) (err error) {
+	// Original code at
+	// <https://github.com/juju/testing/blob/master/checkers/bool.go>.
+	// Copyright 2011 Canonical Ltd.
+	// Licensed under the LGPLv3, see LICENCE file for details.
+	f := reflect.ValueOf(args[0])
+	ftype := f.Type()
+	if ftype.Kind() != reflect.Func || ftype.NumIn() != 1 || ftype.NumOut() != 1 || ftype.Out(0).Kind() != reflect.Bool {
+		return BadCheckf("expected func(T) bool, got %s", ftype)
+	}
+	v, t := reflect.ValueOf(got), ftype.In(0)
+	if !v.IsValid() {
+		if !canBeNil(t.Kind()) {
+			return BadCheckf("provided nil value cannot be used as the argument for %s", ftype)
+		}
+		v = reflect.Zero(t)
+	} else if !v.Type().AssignableTo(t) {
+		return BadCheckf("provided value of type %s cannot be used as the argument for %s", v.Type(), ftype)
+	}
+	result := f.Call([]reflect.Value{v})[0].Interface().(bool)
+	note("result", fmt.Sprint(result))
+	if result {
+		return nil
+	}
+	return fmt.Errorf("value does not satisfy the function")
```

Comment:
"value does not satisfy predicate function"

?

Reply:
Done.